### PR TITLE
Refactor headers of DEM and CFD-DEM

### PR DESCRIPTION
**Pull request by acdaigneault**
# Description of the problem

- Update the year of the copyright of Lethe
- Remove Lethe headers in .cc
- Remove names of authors
- Try to uniformize the header

### CODE REVIEW

File: include/fem-dem/gls_sharp_navier_stokes.h [hunk 1]
```diff
@@ -29,6 +27,9 @@
 
 #include <deal.II/dofs/dof_tools.h>
 
+#ifndef LETHE_GLSSHARPNS_H
+#  define LETHE_GLSSHARPNS_H
+
```
Comment (lpsaavedra) [inline include/fem-dem/gls_sharp_navier_stokes.h:32]: Why was this moved after the #includes? Isn't it generally better to have it before the includes?

File: include/fem-dem/gls_vans.h [hunk 2]
```diff
@@ -45,7 +40,8 @@
 #include <boost/archive/text_iarchive.hpp>
 #include <boost/archive/text_oarchive.hpp>
 
-
+#ifndef lethe_gls_vans_h
+#  define lethe_gls_vans_h
```
Comment (lpsaavedra) [inline include/fem-dem/gls_vans.h:44]: Same as above

File: include/fem-dem/ib_particles_dem.h [hunk 3]
```diff
@@ -35,8 +29,10 @@
 #include <deal.II/dofs/dof_handler.h>
 #include <deal.II/dofs/dof_tools.h>
 
-using namespace dealii;
+#ifndef lethe_ib_particles_dem_h
+#  define lethe_ib_particles_dem_h
```
Comment (lpsaavedra) [inline include/fem-dem/ib_particles_dem.h:34]: Same as above

File: include/fem-dem/parameters_cfd_dem.h [hunk 4]
```diff
@@ -12,20 +12,17 @@
...
-#define lethe_parameters_cfd_dem_h
-
 #include <core/parameters.h>
 #include <core/parameters_lagrangian.h>
 
 #include <deal.II/base/parameter_handler.h>
 #include <deal.II/base/parsed_function.h>
 
+#ifndef lethe_parameters_cfd_dem_h
+#  define lethe_parameters_cfd_dem_h
```
Comment (lpsaavedra) [inline include/fem-dem/parameters_cfd_dem.h:25]: Same as above

File: include/dem/boundary_cells_info_struct.h [hunk 5]
```diff
@@ -1,6 +1,6 @@
 /* ---------------------------------------------------------------------
  *
- * Copyright (C) 2019 - 2019 by the Lethe authors
+ * Copyright (C) 2019 - 2024 by the Lethe authors
```
Comment (lpsaavedra) [inline include/dem/boundary_cells_info_struct.h:3]: There are some files in Lethe that do not have a second date, and are like these:
`Copyright (C) 2019 - by the Lethe authors`
What is the right approach for this? Let's decide on something so we can have the same across all files.

**Review by lpsaavedra — APPROVED** [review]
Nice clean up. We should do this with the whole code... since it is not really uniform.

File: include/fem-dem/gls_sharp_navier_stokes.h (hunk 1 above)
Comment (acdaigneault) [reply include/fem-dem/gls_sharp_navier_stokes.h:32]: I looked into it online and there are no general consensus about it. I think it should be better (with Bruno we looked in deal.II after I did the PR, and yes the includes are in the include guards). I did what was mostly done in Lethe, but I can do the opposite. It could be something in our in-house guidelines.

File: include/dem/boundary_cells_info_struct.h (hunk 5 above)
Comment (acdaigneault) [reply include/dem/boundary_cells_info_struct.h:3]: I agree

Comment (acdaigneault) [reply include/dem/boundary_cells_info_struct.h:3]: I see deal.II uses the same format (year change with the update of the year). 
But I think this is not mandatory, and only the publication date is necessary (legally) https://liferay.dev/blogs/-/blogs/how-and-why-to-properly-write-copyright-statements-in-your-code#why-not-bump-the-year-on-change.
What do you think?

Comment (lpsaavedra) [reply include/dem/boundary_cells_info_struct.h:3]: In that case I guess we can do it in the same way as deal.II

**Review by blaisb — APPROVED** [review]
